### DISCUSSION TITLE
[OvenMediaEnginev0.15.5] Edge crash when play LLHLS stream

### DISCUSSION THREAD

**Discussion by hugmahit** [OvenMediaLabs/OvenMediaEngine#1148] [Help] [answered]
**Describe the bug**
OvenMediaEngine Edge crash when play LLHLS stream

**Expected behavior**
1. Edge: Set Server.xml as follows: [Server.txt](https://github.com/AirenSoft/OvenMediaEngine/files/11029819/Server.txt)
2. OME was started manually by me (instead of using systemd).

**Crash**
File crash: [crash_20230321.dump.txt](https://github.com/AirenSoft/OvenMediaEngine/files/11029806/crash_20230321.dump.txt)

**Server (please complete the following information):**
 - OS: 20.04.1 LTS (GNU/Linux 5.4.0-42-generic) - 48 cores
 - OvenMediaEngine Version: v.0.15.5 [`#f6d8d9d`](https://github.com/AirenSoft/OvenMediaEngine/commit/f6d8d9d9bce839a0b7084d7188b8bab99ad26c6c)
 - Branch: dev

**Additional context**
 - Testing LLHLS stream using 1 Origin - 1 Edge
 - Crash when OME start and has ~ 1200 connection, after restart works fine but sometimes OME down or crash
 - Kernel
![image](https://user-images.githubusercontent.com/43240025/226655478-d48b64b5-4dbf-4c90-9eaa-f5d78452720c.png)
 - Log when crash
![image](https://user-images.githubusercontent.com/43240025/226654065-8704c04f-8fd1-4f63-880d-37f16a5b9f19.png)
 - Sometimes error while playing LLHLS stream
![image](https://user-images.githubusercontent.com/43240025/226656433-43f440c1-209f-439f-9c8f-8ef17000e7cd.png)

**Comment by getroot** [comment]
Thanks for reporting.
Because the crash dump file is stripped, analysis is not easy. Could you please upload the crash dump file created after running it with the debug build binary? If that's difficult, could you upload your debug binary and release binary files?

**Comment by hugmahit** [comment]
This is release binary file: [RELEASE](https://drive.google.com/file/d/1Slog5Cbix1tIo0FpRXiAol2H2JgSSEDa/view?usp=share_link)
Do you need debug binary file. Right now I don't have file.

**Reply by getroot** [reply]
Thank you.  If you give us a debug binary file, it will be very helpful in analyzing the cause of the problem.  You can build a debug file like below.

$ cd ovenmediaengine/src
$ make -j 48
$ ls bin/DEBUG/OvenMediaEngine

**Reply by hugmahit** [reply]
ok @getroot .

**Reply by hugmahit** [reply]
@getroot This is debug binary file: [DEBUG](https://drive.google.com/file/d/190lzTI7mV2N9yKmTro2tKiJzeaS0rDSZ/view?usp=share_link)

**Reply by getroot** [reply]
@hugmahit Thank you very much!!

**Reply by dimiden** [reply]
@hugmahit 
Please attach the ```bin/RELEASE/OvenMediaEngine``` as well. (Sorry for bothering you!)

```
$ cd ovenmediaengine/src
$ make -j $(nproc) release
$ ls bin/RELEASE/OvenMediaEngine
```

Note: The binary generated by ```make release``` command is not stripped.

**Reply by hugmahit** [reply]
No problem if that support you find the problem.
This is release file: [RELEASE](https://drive.google.com/file/d/1pa7BlAa7WdKbn-wA1iJ88KrGTbyryvKq/view?usp=share_link)

**Comment by getroot** [comment, marked as answer]
@hugmahit 
With the information you provided, I have fixed the code I suspect is causing the problem. (6289bb40e934a0dab19c78e4f6eae350067f48f1)
But I can't be sure that this solved your problem as I haven't been able to reproduce it. Please check if your problem is solved with the latest master branch.

**Reply by hugmahit** [reply]
@getroot 
Thank you very much!
If there is any problem I will report back to you

**Reply by getroot** [reply]
It would be very helpful if you could let us know if the problem is not reproduced with the latest version.
